x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtStringClaimValidator.java

```diff
@@ -63,16 +85,14 @@ public void validate(JWSHeader jwsHeader, JWTClaimsSet jwtClaimsSet) {
         if (claimValues == null) {
             throw new IllegalArgumentException("missing required string claim [" + fallbackableClaim + "]");
         }
-
-        if (allowedClaimValues != null && false == claimValues.stream().anyMatch(allowedClaimValues::contains)) {
+        if (false == claimValues.stream().anyMatch(allowedClaimValuesPredicate)) {
```

Comment:
On a second thought, this would not solve it completely. The proper way to short-circuit it would be if `allowedClaimValuesPredicate` accepted a list of strings to check.

Reply:
The `*` used in the `ALLOW_ALL_SUBJECTS` benefits of some short-circuting of sorts: https://github.com/elastic/elasticsearch/blob/5c26cf07622b91166d46ca89900f6fd3ef9c1d05/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/Automatons.java#L214-L216

In the `JwtStringClaimValidator` constructor I could short-circuit the predicate building logic if either one of the "allowed" lists is empty, but I think that would complicate the logic for very little benefit (e.g. I expect the JVM will figure out that set stays empty and expedite the `contains` check).

Reply:
Wasn't aware of a short-circuit in `Automatons`. Agreed, let's keep it simple and optimize if we need to.